File: src/main/java/com/runehive/game/event/impl/ItemOnPlayerEvent.java
package com.runehive.game.event.impl;

import com.runehive.game.event.Event;
import com.runehive.game.world.entity.mob.player.Player;
import com.runehive.game.world.items.Item;

public class ItemOnPlayerEvent implements Event {

    private final Player other;

    private final Item used;

    private final int slot;

    public ItemOnPlayerEvent(Player other, Item used, int slot) {
        this.other = other;
        this.used = used;
        this.slot = slot;
    }

    public Player getOther() {
        return other;
    }

    public Item getUsed() {
        return used;
    }

    public int getSlot() {
        return slot;
    }

}

File: src/main/kotlin/org/jire/runehiveps/event/player/ItemOnPlayerEvent.kt
package org.jire.runehiveps.event.player

import com.runehive.game.event.impl.ItemOnPlayerEvent
import com.runehive.game.plugin.PluginManager
import com.runehive.game.world.World
import com.runehive.game.world.entity.mob.player.Player
import com.runehive.net.packet.out.SendMessage
import org.jire.runehiveps.event.Event
import kotlin.jvm.optionals.getOrNull

/**
 * @author Jire
 */
class ItemOnPlayerEvent(
    val interfaceId: Int,
    val item: Int,
    val itemSlot: Int,
    val slot: Int
) : Event {

    override fun handle(player: Player) {
        val used = player.inventory[itemSlot] ?: return
        if (!used.matchesId(item)) return

        val other = World.getPlayerBySlot(slot).getOrNull() ?: return
        
        player.walkTo(other) {
            player.face(other.position)
            if (!PluginManager.getDataBus().publish(player, ItemOnPlayerEvent(other, used, itemSlot))) {
                player.send(SendMessage("Nothing interesting happens."))
            }
        }
    }

}